app/src/main/java/org/simple/clinic/navigation/v2/fragments/BaseScreen.kt

```diff
@@ -0,0 +1,112 @@
+package org.simple.clinic.navigation.v2.fragments
+
+import android.os.Bundle
+import android.os.Parcelable
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.fragment.app.Fragment
+import com.spotify.mobius.Connection
+import com.spotify.mobius.EventSource
+import com.spotify.mobius.Init
+import com.spotify.mobius.MobiusLoop
+import com.spotify.mobius.Next.noChange
+import com.spotify.mobius.Update
+import com.spotify.mobius.android.MobiusAndroid
+import com.spotify.mobius.extras.Connectables
+import com.spotify.mobius.functions.Consumer
+import com.spotify.mobius.rx2.RxMobius
+import io.reactivex.Observable
+import io.reactivex.ObservableTransformer
+import org.simple.clinic.util.unsafeLazy
+import org.simple.clinic.mobius.eventSources
+import org.simple.clinic.mobius.first
+import org.simple.clinic.navigation.v2.ScreenKey
+
+abstract class BaseScreen<K: ScreenKey, M: Parcelable, E, F>: Fragment() {
+
+  companion object {
+    private const val KEY_MODEL = "org.simple.clinic.navigation.v2.fragments.BaseScreen.KEY_MODEL"
+  }
+
+  private val loop: MobiusLoop.Builder<M, E, F> by unsafeLazy {
+    RxMobius
+        .loop(createUpdate()::update, createEffectHandler())
+        .init(createInit())
+        .eventSources(additionalEventSources())
+  }
+
+  private val controller: MobiusLoop.Controller<M, E> by unsafeLazy {
+    MobiusAndroid.controller(loop, defaultModel())
+  }
+
+  protected val screenKey by unsafeLazy { ScreenKey.key<K>(this) }
+
+  @LayoutRes
+  abstract fun layoutResId(): Int
```

Comment:
Since we are providing the binding generic in next commit, we can make the binding as constructor parameter and pass `binding.root` to the `FragmentConstructor`.

Reply:
We cannot pass the binding as a constructor parameter because that is not something that can be saved and restored, so we won't be able to add a default constructor. This is also why we cannot pass the layout res ID as a constructor parameter because this will stop us from being able to construct a view binding instance, right?

Reply:
Ah right, looks like we can only pass `LayoutResId` via the `Fragment` constructor. We can stick with existing approach then 👍🏽